src/main/kotlin/com/nikitanazarov/coroutinestacks/CoroutineFramesList.kt

```diff
@@ -0,0 +1,133 @@
+        setListData(data.toTypedArray())
+        val lastStackFrame = data.getOrNull(1)
+
+        border = if (lastRunningStackFrame == lastStackFrame) {
+            createRoundedBorder(JBColor.BLUE)
+        } else {
+            createRoundedBorder(JBColor.BLACK)
+        }
+
+        cellRenderer = CustomCellRenderer(trace.coroutinesActiveLabel)
+        addMouseListener(object : MouseAdapter() {
+            override fun mouseClicked(e: MouseEvent?) {
+                val list = e?.source as? JBList<*> ?: return
+                val index = list.locationToIndex(e.point).takeIf { it > 0 } ?: return
+                val stackFrameItem = trace.stackFrameItems[index - 1]
+
+                stackFrameItem?.let { frameItem ->
+                    val frame = frameItem.createFrame(debugProcess)
+
+                    if (activeExecutionStack != null && frame != null) {
+                        suspendContext.setCurrentStackFrame(activeExecutionStack, frame)
+                    }
+                }
+            }
+        })
+    }
+
+    // Copied from org.jetbrains.kotlin.idea.debugger.coroutine.view.CoroutineSelectedNodeListener#setCurrentStackFrame
+    private fun SuspendContextImpl.setCurrentStackFrame(executionStack: XExecutionStack?, stackFrame: XStackFrame) {
+        val fileToNavigate = stackFrame.sourcePosition?.file ?: return
+        val session = debugProcess.session.xDebugSession ?: return
+        if (!ClsClassFinder.isKotlinInternalCompiledFile(fileToNavigate)) {
+            ApplicationManager.getApplication().invokeLater {
+                if (executionStack != null) {
+                    session.setCurrentStackFrame(executionStack, stackFrame, false)
+                }
+            }
+        }
+    }
+
+    private fun createRoundedBorder(color: JBColor): Border {
+        val cornerRadius = Constants.cornerRadius
+        val borderThickness = Constants.borderThickness
+
+        val roundedBorder = object : LineBorder(color, borderThickness) {
+            override fun getBorderInsets(c: Component?): Insets {
+                val insets = super.getBorderInsets(c)
+                return JBUI.insets(insets.top, insets.left, insets.bottom, insets.right)
+            }
+
+            override fun paintBorder(c: Component?, g: Graphics?, x: Int, y: Int, width: Int, height: Int) {
+                val g2d = g as? Graphics2D ?: return
+                val arc = 2 * cornerRadius
+                g2d.setRenderingHint(RenderingHints.KEY_ANTIALIASING, RenderingHints.VALUE_ANTIALIAS_ON)
+                g2d.color = color
+                g2d.drawRoundRect(x, y, width - 1, height - 1, arc, arc)
+            }
+        }
+
+        return roundedBorder
+    }
+
+    class CustomCellRenderer(
+        private val coroutinesActive: String?,
+        padding: Int = 3
+    ) : DefaultListCellRenderer() {
+        private val itemBorder = BorderFactory.createMatteBorder(0, 0, 1, 0, JBColor.GRAY)
```

Comment:
these 3 variables clearly belong to a companion object in the CoroutineFramesListClass